knife/lib/chef/knife/bootstrap.rb

```diff
@@ -974,13 +974,17 @@ def ssh_identity_opts
           # Reference: https://github.com/chef/chef/blob/master/lib/chef/knife/ssh.rb#L272
           opts[:keys_only] = config.key?(:connection_password) == false
         else
-          opts[:key_files] = []
+          opts[:key_files] = nil
```

Comment:
What was this intended to fix?   

In train, a value of `nil` for `key_files` will get converted to `[]`.  Given that, this change and the one below /should/ have a net of  no effect.

Reply:
@marcparadise, in train `nil` for `key_files` is not converting to `[]`.
For reference: https://github.com/inspec/train/blob/62a8b1e0b5f6e2cc88b0ab2e4e9101c571bc99df/lib/train.rb#L179

Reply:
That is used for train's CLI usage, IIRC; that is not the path we take.

Our path starts in `TrainConnector.connection`, which invokes `Train.create`[1]  This returns a new instance of  `Train::Transports::SSH` [2].   We then call `instance.connection` to retrieve the connection object.  `connection` calls `validate_options` [3] which in turn initializes `key_files` as `Array(options[:key_files])`[4]. When `key_files` is nil, that will evaluate to `[]`. 

[1] https://github.com/chef/chef/blob/master/knife/lib/chef/knife/bootstrap/train_connector.rb#L54
[2] https://github.com/inspec/train/blob/master/lib/train.rb#L20
[3] https://github.com/inspec/train/blob/master/lib/train/transports/ssh.rb#L79. 
[4] https://github.com/inspec/train/blob/master/lib/train/transports/ssh.rb#L109